File: data/code_pairs/fortran/1861.f90
program test_integration
    implicit none
    double precision :: result, error
    double precision, external :: f
    double precision, parameter :: a = 0.0d0, b = 1.0d0
    double precision, parameter :: epsabs = 1.0e-10, epsrel = 1.0e-10

    ! Assuming qag is a subroutine from QUADPACK
    ! This is a mock-up for demonstration
    call qag(f, a, b, epsabs, epsrel, result, error)
    print *, 'Integration result: ', result
    print *, 'Expected result: ', 1.0 / 3.0
    print *, 'Estimated error: ', error

contains

    ! Mock implementation of qag, for demonstration
    subroutine qag(f, a, b, epsabs, epsrel, result, error)
        implicit none
        double precision, intent(in) :: a, b, epsabs, epsrel
        double precision, external :: f
        double precision, intent(out) :: result, error
        double precision :: h, fa, fb, m, fm, i1, i2
        
        h = b - a
        fa = f(a)
        fb = f(b)
        m = (a + b) / 2.0d0
        fm = f(m)
        
        i1 = h * (fa + 4.0d0*fm + fb) / 6.0d0
        i2 = h * (fa + fb) / 2.0d0
        
        result = i1
        error = abs(i1 - i2)
    end subroutine qag

    ! Function to integrate
    double precision function f(x)
        double precision, intent(in) :: x
        f = x**2
    end function f

end program test_integration

File: data/code_pairs/cpp/1861.cpp
#include <iostream>
#include <cmath>
#include <functional>

class SimpleQuadpack {
public:
    double integrate(std::function<double(double)> f, double a, double b, int n) {
        double h = (b - a) / n;
        double s = f(a) + f(b);
        for (int i = 1; i < n; i += 2) {
            s += 4 * f(a + i * h);
        }
        for (int i = 2; i < n - 1; i += 2) {
            s += 2 * f(a + i * h);
        }
        return s * h / 3;
    }
};

// Function to integrate: f(x) = x^2
double f(double x) {
    return x * x;
}

int main() {
    SimpleQuadpack qp;
    double a = 0.0; // Lower limit of integration
    double b = 1.0; // Upper limit of integration
    int n = 1000; // Number of subdivisions for the integration
    
    double result = qp.integrate(f, a, b, n);
    std::cout << "Integration result: " << result << std::endl;
    std::cout << "Expected result: " << 1.0 / 3.0 << std::endl;
    
    return 0;
}